main.md

```diff
@@ -193,180 +199,327 @@ Note: Applications MUST ensure that their authorization data types do not collid
+The AS will typically also add further claims to the JWT the RS requires for request processing, e.g., user id, roles, and transaction specific data. What claims the particular RS requires is defined by the RS-specific policy with the AS.
 
-The `locations` and the `identifier` elements together allow the AS to determine the resource a client wants to access as shown in following example: 
+The following shows the contents of an example JWT payload for the payment initation example above: 
 
 ```JSON
-[  
-   {
-      "type": "https://scheme.example.org/storage":
-      "locations":["https://storage.example.com"],
-      "identifier":"/shared/group1",
-      "actions":[  
-         "read"
-      ]
-   }
+{
+   "iss": "https://as.example.com",
+   "sub": "24400320",
+   "aud": "s6BhdRkqt3",
+   "exp": 1311281970,
+   "acr": "psd2_sca",
+   "txn": "8b4729cc-32e4-4370-8cf0-5796154d1296",
+   "authorization_details": [
+      {
+         "type": "https://www.someorg.com/payment_initiation",
+         "actions": [
+            "initiate",
+            "status",
+            "cancel"
+         ],
+         "locations": [
+            "https://example.com/payments"
+         ],
+         "instructedAmount": {
+            "currency": "EUR",
+            "amount": "123.50"
+         },
+         "creditorName": "Merchant123",
+         "creditorAccount": {
+            "iban": "DE02100100109307118603"
+         },
+         "remittanceInformationUnstructured": "Ref Number Merchant"
+      }
+   ]
 }
 ```
-The AS MUST respect those values when deciding whether a certain element is placed into a (structured) access token or token introspection response.
 
+In this case, the AS added the following example claims:
+
+* `sub`: conveys the user on which behalf the client is asking for payment initation
+* `txn`: transaction id used to trace the transaction across the services of provider `example.com`
+* `debtorAccount`: API-specific element containing the debtor account as selected by the user during the authorization process. The field `user_role` conveys the role the user has with respect to this particuar account. In this case, she is the owner. This data is used for access control at the payment API (the RS).
+
+## Token Introspection Request
+
+In case of opaque access tokens, the data provided to a certain RS is determined using the RS's identifier with the AS (see [@I-D.ietf-oauth-jwt-introspection-response], section 3). 
+
+## Token Introspection Response
+
+The token introspection response provides the RS with the authorization details applicable to it as a top-level JSON element along with the claims the RS requires for request processing. 
+
+Here is an example for the payment initation example RS:
+
+```json
+{
+   "active": true,
+   "sub": "24400320",
+   "aud": "s6BhdRkqt3",
```

Comment:
the client id used throughout the examples elsewhere in the document is the audience here?

Reply:
fixed